src/core/atomic.d

```diff
@@ -1143,6 +1143,45 @@ version (CoreUnittest)
         }
     }
 
+    @betterC pure nothrow @nogc unittest
+    {
+        byte[] byteArray = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19];
+        shared(ulong)[] ulongArray = [2, 4, 6, 8, 10, 12, 14, 16, 19, 20];
+
+        {
+            auto array = byteArray;
+            byte* ptr = &array[0];
+            byte* prevPtr = atomicFetchAdd(ptr, 3);
+            assert(prevPtr == &array[0]);
+            assert(*prevPtr == 1);
+            assert(*ptr == 7);
+        }
+        {
+            auto array = cast(shared(typeof(byteArray))) byteArray;
+            shared byte* ptr = &array[0];
+            shared(byte)* prevPtr = atomicFetchAdd(ptr, 3);
+            assert(prevPtr == &array[0]);
+            assert(*prevPtr == 1);
+            assert(*ptr == 7);
+        }
+        {
+            auto array = ulongArray;
+            ulong* ptr = &array[0];
```

Comment:
How does this compile? `shared(ulong)*` assigned to `ulong*` should be an error? :/

Reply:
It doesn't. This is a result of the forced change 
        - ulong[] ulongArray = [2, 4, 6, 8, 10, 12, 14, 16, 19, 20];
        + shared(ulong)[] ulongArray = [2, 4, 6, 8, 10, 12, 14, 16, 19, 20];

Reply:
Did I mis-click something? I saw the suggesting in the review (I've never seen that in github before), and I replied to the suggestion... I didn't think I clicked anything that applied the suggestion though, but maybe I fat-fingered it?

Reply:
Not sure. Right now it doesn't pass CI for some reason and I don't even know how to begin debugging this. I know my change (without unit tests) did work on master. I failed to rebase and this is why I ended up with a new pull request.